src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java

```diff
@@ -28,15 +47,70 @@ public WarehouseDto findById(Long warehouseId) {
                 .orElseThrow(() -> new ResourceNotFoundException("Warehouse with id=" + warehouseId + " doesn't exist"));
     }
 
-    // TODO: 11/4/20
+    @Transactional
     @Override
-    public WarehouseDto save(WarehouseDto dto) {
-        return null;
+    public WarehouseDto save(Long customerId, WarehouseDto warehouseDto) {
+        if (warehouseDto.getIdentifier() == null) {
+            throw new ResponseStatusException(
+                    HttpStatus.NOT_ACCEPTABLE, "\"Identifier\" can't be null");
+        }
+
+        Warehouse warehouse = warehouseMapper.map(warehouseDto);
+        Address savedAddress = addressCreate(warehouseDto.getAddressDto());
+        warehouse.setAddress(savedAddress);
```

Comment:
How can you create warehouse with non-existing address? Address should be pre-created, shouldn't it?